test/disjointPoolFileProv.cpp

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (C) 2023-2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <random>
+
+#include <umf/pools/pool_disjoint.h>
+#include <umf/providers/provider_file_memory.h>
+
+#include "coarse.h"
+#include "provider.hpp"
+
+using umf_test::KB;
+using umf_test::MB;
+using umf_test::test;
+
+#define FILE_PATH ((char *)"tmp_file")
+
+umf_memory_provider_ops_t UMF_MALLOC_MEMORY_PROVIDER_OPS =
+    umf::providerMakeCOps<umf_test::provider_ba_global, void>();
+
+struct FileWithMemoryStrategyTest
+    : umf_test::test,
+      ::testing::WithParamInterface<coarse_strategy_t> {
+    void SetUp() override {
+        test::SetUp();
+        allocation_strategy = this->GetParam();
+    }
+
+    coarse_strategy_t allocation_strategy;
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    FileWithMemoryStrategyTest, FileWithMemoryStrategyTest,
+    ::testing::Values(UMF_COARSE_MEMORY_STRATEGY_FASTEST,
+                      UMF_COARSE_MEMORY_STRATEGY_FASTEST_BUT_ONE,
+                      UMF_COARSE_MEMORY_STRATEGY_CHECK_ALL_SIZE));
+
+TEST_P(FileWithMemoryStrategyTest, disjointFileMallocPool_simple1) {
+    umf_memory_provider_handle_t malloc_memory_provider;
```

Comment:
Some declarations are assigned to NULL, others not. Can it make Coverity scream about using variables without assignment?

Reply:
I cannot predict what Coverity can detect. 
I have initialized all `malloc_memory_provider` to `nullptr`.
Done